### PR TITLE
Check for uncommitted files beyond the `diff` directory

### PR DESCRIPTION
**Pull request by jeffwidman**
This removes the indirection of the bash script so that it's immediately clear from the github action definition file what is being checked. Currently the logic is split between the action which does the rebuild and the bash script which checks the diff.

It also checks for _any_ delta in the git repo, not just the `dist/` directory. Any change should fail CI until it's either committed or added to `.gitignore`.

### CODE REVIEW

File: .github/workflows/check-uncommitted.yml [hunk 1]
```diff
@@ -31,5 +32,7 @@ jobs:
       - name: Rebuild the dist/ directory
         run: npm run build
 
-      - name: Compare the expected and actual dist/ directories
-        run: bin/check-diff
```
Comment (brrygrdn) [inline .github/workflows/check-uncommitted.yml:35]: As I recall, the intent of using this script was to have the Check Run show the diff on failure so you didn't need to run it on your machine to see what the issue was.

This was useful as we had environment-specific issues early on where the Typescript compile wasn't properly encapsulated and produced a diff on Actions, but none locally.

> It also checks for any delta in the git repo, not just the dist/ directory. Any change should fail CI until it's either committed or added to .gitignore.

That's fair - I don't have a strong opinion on this, I think it was originally just the dist/ directory to keep the CI build focused on verifying the distributed code as any other changes were less impactful.

File: bin/check-diff [hunk 2]
```diff
@@ -1,11 +0,0 @@
-#!/bin/bash
-
-# Make sure we notice any untracked files generated by the build
-git add --intent-to-add .
-git diff --quiet dist/
-retVal=$?
-if [ $retVal -ne 0 ]; then
-  echo "Detected uncommitted changes after build:"
-  git --no-pager diff dist/
```
Comment (brrygrdn) [inline bin/check-diff:9]: The contents of the `diff/` folder are marked as [generated](https://github.com/dependabot/fetch-metadata/blob/main/.gitattributes) so this was just to ensure we spat out the actual change in the obfuscated JS.

**Comment by jeffwidman** [comment]
Hmm... I'm really not sure what to do here, as @brrygrdn raises good points, but I'm still not excited by having a separate script... that said, movign the `bin/check-diff` code in seems a bit complicated. And probably best to ensure we _do_ check all files, not just `dist/`... 

I'll let this float a bit long in case anyone else has opinions, otherwise I'll probably either close this or leave existing code as-is, but add a follow-on step so that if there are no changes in `dist/` to ensure no other uncommitted changes elsewhere either... 

I guess what also could be helpful here is a few more code comments in the source so that someone down the road doesn't open a PR on this either.

File: bin/check-build-output-in-dist-directory
Comment (jeffwidman) [inline bin/check-build-output-in-dist-directory:1]: Code didn't change in this script, only renamed it and added code comments for improved clarity.

File: .github/workflows/check-uncommitted.yml (hunk 1 above)
Comment (jeffwidman) [reply .github/workflows/check-uncommitted.yml:31]: This is useful, so I kept it, and then augmented with another job that looks for any other uncommitted files.

**Comment by jeffwidman** [comment]
I updated based on @brrygrdn 's helpful context/feedback.

This is ready for another review.

**Review by brrygrdn — APPROVED** [review]
🚀

**Pull request merged**